pootle/checks.py

```diff
@@ -0,0 +1,158 @@
+from django.core import checks
+from django.utils.translation import ugettext as _
+
+
+# Minimum Translate Toolkit version required for Pootle to run.
+TTK_MINIMUM_REQUIRED_VERSION = (1, 11, 0)
+
+# Minimum Django version required for Pootle to run.
+DJANGO_MINIMUM_REQUIRED_VERSION = (1, 7, 5)
+
+# Minimum lxml version required for Pootle to run.
+LXML_MINIMUM_REQUIRED_VERSION = (2, 2, 2, 0)
+
+# Minimum Redis server version required.
+# Initially set to some minimums based on:
+# 1. Ubuntu 12.04LTS's version 2.8.4 (10.04LTS was too old for RQ)
+# 2. RQ requires >= 2.6.0, and
+# 3. Wanting to insist on at least the latest stable that devs are using i.e.
+#    2.8.* versions of Redis
+REDIS_MINIMUM_REQUIRED_VERSION = (2, 8, 4)
+
+
+@checks.register()
+def test_library_versions(app_configs, **kwargs):
+    from django import VERSION as django_version
+    from lxml.etree import LXML_VERSION as lxml_version
+    from translate.__version__ import ver as ttk_version
+
+    errors = []
+
+    if django_version < DJANGO_MINIMUM_REQUIRED_VERSION:
+        errors.append(checks.Critical("Your version of Django is too old.",
+            hint="Try pip install --upgrade Django",
+            id="pootle.E004",
+        ))
+
+    if lxml_version < LXML_MINIMUM_REQUIRED_VERSION:
+        errors.append(checks.Warning("Your version of lxml is too old.",
+            hint="Try pip install --upgrade lxml",
+            id="pootle.E005",
+        ))
+
+    if ttk_version < TTK_MINIMUM_REQUIRED_VERSION:
+        errors.append(checks.Critical("Your version of Translate Toolkit is too old.",
+            hint="Try pip install --upgrade translate-toolkit",
+            id="pootle.E006",
```

Comment:
IMHO these three should be critical, not regular errors. Or at least the Django and TTK ones.